packages/sit-onyx/src/components/OnyxPageLayout/OnyxPageLayout.vue

```diff
@@ -17,31 +17,27 @@ const slots = defineSlots<{
 
 const pageModifier = computed(() => {
   let mode = "";
-  if (!slots.footer && slots.sidebar) mode = "onyx-page--side-main";
-  if (slots.footer && (!slots.sidebar || props.hideSidebar)) mode = "onyx-page--main-footer";
-  if (slots.footer && slots.sidebar) {
-    if (props.footerAsideSidebar) {
-      mode = "onyx-page--side-main-footer-partial";
-    } else {
-      mode = "onyx-page--side-main-footer-full";
-    }
+  if (props.footerAsideSidebar) {
+    mode = "onyx-page--footer-partial";
+  } else {
+    mode = "onyx-page--footer-full";
   }
   return mode;
 });
 </script>
 
 <template>
   <div class="onyx-page" :class="pageModifier">
-    <aside v-if="slots.sidebar && !props.hideSidebar" class="onyx-page__sidebar">
+    <aside v-show="slots.sidebar && !props.hideSidebar" class="onyx-page__sidebar">
```

Comment:
I would prefer to not render the sidebar/footer if its not set, otherwise it might be weird if it shows in the dev tools although it is not needed.

Can you extend why the slot switching is not stable in the first place? Slots and properties are reactive so I assume the issue is related to the application, not onyx itself

Reply:
The computed in line 18 `pageModifier` did not get updated with the described steps to reproduce.

But actually, with the change that it now only depends on 1 prop, it's stable already - we can keep v-if 🎉